pyqmc/reblock.py

```diff
@@ -0,0 +1,55 @@
+import pyblock
+import pandas as pd
+import numpy as np
+from scipy.stats import sem
+
+
+def reblock(data, reblocks, col):
+    '''
+        Reblocks data according to “Error estimates on averages of correlated data”,
+        H. Flyvbjerg, H.G. Petersen, J. Chem. Phys. 91, 461 (1989).
+    '''
+    edat = data[col].values
+    n = len(edat)
+    for i in range(reblocks):
+        edat_prime = []
+        for j in range(1, int(len(edat)/2+1)):
+            edat_prime.append((edat[2*j-2] + edat[2*j-1])/2)
+        edat = edat_prime
+    return np.array(edat)
+
+
+def optimally_reblocked(data, cols):
```

Comment:
Could this be simplified? For example: 
def optimal_reblock(data):
     reblocked_data=pyblock.pd_utils.reblock(data)
     return pyblock.pd_utils.summary(reblocked_data[1])

This would be called as 
optimally_reblocked(data[['col1','col2',...]])

Reply:
I tried to do this but when different runs have different optimal reblocks it breaks down. For example, I used dmc.py to generate data. energytotal, energyee, energyei were optimally reblocked after 9 iterations, were energyke was optimally reblocked after 8. When I computed it in the way you suggested, it returned that all runs were optimally reblocked after 9 iterations. This slightly changed the final mean for energyke and caused it to fail the test I wrote.

Reply:
The mean can change a little when reblocking, since some data will be thrown out. That is not an issue.